base_partner_merge_user_consolidation/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2025 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        """Merge partners, assign a single user"""
+        super()._merge(partner_ids, dst_partner=dst_partner, extra_checks=extra_checks)
+        # Recompute dst_partner
+        Partner = self.env["res.partner"]
+        partner_ids_rs = Partner.browse(partner_ids).exists()
+        if not partner_ids_rs or len(partner_ids_rs) < 1:
+            return
+        if dst_partner and dst_partner.exists():
+            final_partner = dst_partner
+        else:
+            final_partner = self._get_ordered_partner(partner_ids_rs.ids)[-1]
+        # Safety: final partner might have been deleted if something extreme happened,
+        # but in normal core flow it's the surviving one.
+        if not final_partner or not final_partner.exists():
+            return
+        # Find all users for final partner
+        Users = self.env["res.users"].sudo()
+        partner_users = Users.search([("partner_id", "=", final_partner.id)])
+        if len(partner_users) <= 1:
+            return
+        kept_user, losing_users = self._pick_kept_user(partner_users)
+        # Link and make sure that user is active
+        kept_user.write(
+            {
+                "partner_id": final_partner.id,
+                "active": True,
+            }
+        )
+        # Merge all groups from all users to final user
+        self._union_user_groups(kept_user, losing_users)
+        # Deactivate remaining users
+        self._archive_users(losing_users)
+
+    def _pick_kept_user(self, users):
+        """Find user by login_date or create_date"""
+        if not users:
```

Comment:
This cannot happen, as you only call this after checking number of users. Remove test, and this will increase coverage percentage as well.